tests/pipeline/test_node.py

```diff
@@ -539,12 +539,25 @@ def test_valid_multi_level_namespace(self):
         assert n.inputs == ["namespace.subnamespace.input_dataset"]
         assert n.outputs == ["namespace.subnamespace.output_dataset"]
 
-    def test_invalid_multi_level_namespace(self):
-        """Test that mismatched multi-level namespaces raise a ValueError."""
-        with pytest.raises(ValueError, match="Invalid dataset name"):
+    def test_only_top_level_namespace_match(self):
```

Comment:
maybe the test name here can be  `test_dataset_namespace_matches_pipeline_namespace(self)`: